From 2ed829c7fdd2f31290094c1fe83267c7ca82330c Mon Sep 17 00:00:00 2001
From: Gunar Schorcht <gunar@schorcht.net>
Date: Thu, 10 Mar 2022 12:38:51 +0100
Subject: [PATCH 18/20] bootloader: changes for esp32c3

Since RIOT directly links compiled objects, the archive/object file references must be replaced with the corresponding object file paths.
---
 .../subproject/main/ld/esp32c3/bootloader.ld  | 55 ++++++++++---------
 1 file changed, 28 insertions(+), 27 deletions(-)

diff --git a/components/bootloader/subproject/main/ld/esp32c3/bootloader.ld b/components/bootloader/subproject/main/ld/esp32c3/bootloader.ld
index c9b2da7612c..18b939322ba 100644
--- a/components/bootloader/subproject/main/ld/esp32c3/bootloader.ld
+++ b/components/bootloader/subproject/main/ld/esp32c3/bootloader.ld
@@ -23,34 +23,35 @@ SECTIONS
     _loader_text_start = ABSOLUTE(.);
     *(.stub .gnu.warning .gnu.linkonce.literal.* .gnu.linkonce.t.*.literal .gnu.linkonce.t.*)
      *(.iram1 .iram1.*) /* catch stray IRAM_ATTR */
-    *liblog.a:(.literal .text .literal.* .text.*)
+    *components/log/*(.literal .text .literal.* .text.*)
     *libgcc.a:(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:bootloader_clock_loader.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:bootloader_common_loader.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:bootloader_flash.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:bootloader_random.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:bootloader_random*.*(.literal.bootloader_random_disable .text.bootloader_random_disable)
-    *libbootloader_support.a:bootloader_random*.*(.literal.bootloader_random_enable .text.bootloader_random_enable)
-    *libbootloader_support.a:bootloader_efuse_esp32c3.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:bootloader_utility.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:bootloader_sha.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:bootloader_console_loader.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:bootloader_panic.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:bootloader_soc.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:esp_image_format.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:flash_encrypt.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:flash_encryption_secure_features.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:flash_partitions.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:secure_boot.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:secure_boot_secure_features.*(.literal .text .literal.* .text.*)
-    *libbootloader_support.a:secure_boot_signatures_bootloader.*(.literal .text .literal.* .text.*)
-    *libmicro-ecc.a:*.*(.literal .text .literal.* .text.*)
-    *libspi_flash.a:*.*(.literal .text .literal.* .text.*)
-    *libhal.a:wdt_hal_iram.*(.literal .text .literal.* .text.*)
-    *libesp_hw_support.a:rtc_clk.*(.literal .text .literal.* .text.*)
-    *libesp_hw_support.a:rtc_time.*(.literal .text .literal.* .text.*)
-    *libesp_hw_support.a:regi2c_ctrl.*(.literal .text .literal.* .text.*)
-    *libefuse.a:*.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/bootloader_clock_loader.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/bootloader_common_loader.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/bootloader_flash.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/bootloader_random.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/bootloader_random*.*(.literal.bootloader_random_disable .text.bootloader_random_disable)
+    *components/bootloader_support/*/bootloader_random*.*(.literal.bootloader_random_enable .text.bootloader_random_enable)
+    *components/esp_common/src/fpga_overrides.*(.literal.bootloader_fill_random .text.bootloader_fill_random)
+    *components/bootloader_support/*/bootloader_efuse_esp32c3.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/bootloader_utility.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/bootloader_sha.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/bootloader_console_loader.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/bootloader_panic.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/bootloader_soc.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/esp_image_format.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/flash_encrypt.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/flash_encryption_secure_features.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/flash_partitions.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/secure_boot.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/secure_boot_secure_features.*(.literal .text .literal.* .text.*)
+    *components/bootloader_support/*/secure_boot_signatures_bootloader.*(.literal .text .literal.* .text.*)
+    *components/*/micro-ecc/*(.literal .text .literal.* .text.*)
+    *components/spi_flash/*(.literal .text .literal.* .text.*)
+    *components/hal/wdt_hal_iram.*(.literal .text .literal.* .text.*)
+    *components/esp_hw_support/*/rtc_clk.*(.literal .text .literal.* .text.*)
+    *components/esp_hw_support/*/rtc_time.*(.literal .text .literal.* .text.*)
+    *components/esp_hw_support/*/regi2c_ctrl.*(.literal .text .literal.* .text.*)
+    *components/efuse/*(.literal .text .literal.* .text.*)
     *(.fini.literal)
     *(.fini)
     *(.gnu.version)
-- 
2.17.1

